core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -610,10 +617,21 @@ private Sha256Hash calculateMerkleRoot() {
         return tree;
     }
 
-    private void checkTransactions() throws VerificationException {
+    /**
+     * Verify the transactions on a block.
+     *
+     * @param height block height, if known. If provided, used to validate the
+     * coinbase input script of v2 and above blocks.
+     * @throws VerificationException if there was an error verifying the block.
+     */
+    private void checkTransactions(final Integer height, final Set<VerificationFlags> flags)
```

Comment:
Or just use a traditional int with a bitfield. Set<Flag> is quite a lot of overhead.